packages/components/src/components/tooltip/tooltip.tsx

```diff
@@ -0,0 +1,263 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  Method,
+  Prop,
+  Watch,
+  h,
+  State,
+} from '@stencil/core';
+import Popover from './utilities/popover';
+
+let id = 0;
+
+@Component({
+  tag: 'scale-tooltip',
+  styleUrl: 'tooltip.css',
+  shadow: true,
+})
+export class Tooltip {
+  componentId = `tooltip-${++id}`;
+  isVisible = false;
+  popover: Popover;
+  tooltipPositioner: HTMLElement;
+  target: HTMLElement;
+  tooltip: any;
+
+  @Element() host: HTMLElement;
+  /** (optional) The content of the Tooltip supporting Text only */
+  @Prop() content = '';
+  /** (optional) Position of the Tooltip on the Object */
+  @Prop() placement:
+    | 'top'
+    | 'top-start'
+    | 'top-end'
+    | 'right'
+    | 'right-start'
+    | 'right-end'
+    | 'bottom'
+    | 'bottom-start'
+    | 'bottom-end'
+    | 'left'
+    | 'left-start'
+    | 'left-end' = 'top';
+  /** (optional) Disable Tooltip */
+  @Prop() disabled = false;
+  /** (optional) Distance of the Tooltip from the Target Object (related to the `placement`) */
+  @Prop() distance = 5;
+  /** (optional) Set the Tooltip to open per default (will still be closed on closing Events) */
+  @Prop({ mutable: true, reflect: true }) open = false;
+  /** (optional) skidding moves the tooltip of the element in dependence of its `placement` to the element either on an x-axis (at `placement` top/down)
+  or on a y-axis (for output `placement` left/right) */
+  @Prop() skidding = 0;
+  /** (optional) Set custom trigger Event selection */
+  @Prop() trigger: string = 'hover focus';
+  /** (optional) Switching the flip option of the tooltip on and off*/
+  @Prop() flip: boolean = true;
+  /** (optional) Switching the preventOverflow option of the tooltip on and off*/
+  @Prop() preventOverflow: boolean = false;
+
+  @State() mouseOverTooltip: boolean = false;
+
+  @Watch('open')
+  handleOpenChange() {
+    this.open ? this.showTooltip() : this.hideTooltip();
+  }
+
+  @Event({ eventName: 'scale-show' }) tooltipShow: EventEmitter;
+  @Event({ eventName: 'scale-aftershow' }) tooltipAfterShow: EventEmitter;
+  @Event({ eventName: 'scale-hide' }) tooltipHide: EventEmitter;
+  @Event({ eventName: 'scale-after-hide' }) tooltipAfterHide: EventEmitter;
```

Comment:
To follow the same pattern we have in the Modal, I would rename these events:

- `scale-show` → `scale-before-show`
- `scale-aftershow` → `scale-show`
- `scale-hide` → `scale-before-hide`
- `scale-after-hide` → `scale-hide`

This way to can `preventDefaultp the action with the "before" event, and you get the final "show" or "hide" events after they actually happen. Does that make sense to you?